File: fabric-frontend/src/app/api/models/query-composite-dto.ts
/* tslint:disable */
/* eslint-disable */
import { QueryDto } from './query-dto';
import { ResultDto } from './result-dto';
export interface QueryCompositeDto {
  children?: Array<QueryCompositeDto>;
  id?: string;
  query?: QueryDto;
  result?: ResultDto;
}

File: fabric-backend-core/src/main/java/io/storydoc/fabric/query/app/composite/QueryCompositeDTO.java
package io.storydoc.fabric.query.app.composite;

import io.storydoc.fabric.query.app.QueryDTO;
import io.storydoc.fabric.query.app.ResultDTO;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
class QueryCompositeDTO {

    private String id;
    private QueryDTO query;
    private ResultDTO result;

    private List<QueryCompositeDTO> children;
}
